include/onnxruntime/core/framework/tensor.h

```diff
@@ -50,6 +50,20 @@ using BufferNakedPtr = void*;
 #pragma GCC diagnostic ignored "-Wnull-dereference"
 #endif
 #endif
+
+/**
+ * @brief Create a closure for releasing a buffer
+ * @param alloc   allocator for buffer releasing
+ * @param p_data  start addr of the buffer
+ * @return  lambda function for releasing the buffer
+*/
+inline std::function<void(void)> CreateBufDelClr(AllocatorPtr&& alloc, void* p_data) {
+  AllocatorPtr deleter(std::move(alloc));
```

Comment:
Given the data in a Tensor should only ever come from a single AllocatorPtr, do we need this, or could we just instead use what was Tensor::buffer_deleter_?

Reply:
I think we may have to do more than one things when deleting a Tensor, and just using the buffer_deleter_ is not enough.

It's the complex function TensorProtoToMLValue. As we discuss before it takes a pre-allocated MemBuffer. Further more, it appears to me sometimes this function create a deleter function that have to be called when the pre-allocated MemBuffer is released. In some cases this extra deleter must be called before the MemBuffer is freed.

So in some cases we do have more than one thing to do here. If I understand correctly, when file memory mapping is enabled, and tensor raw data is in a separated file, the Tensor occupies two separate buffers and both of them have to be released.

This is created so that I can stick to the original behavior as much as possible.